overlord/ifacestate/helpers.go

```diff
@@ -107,7 +107,7 @@ func (m *InterfaceManager) addSnaps() error {
 	for _, snapInfo := range snaps {
 		addImplicitSlots(snapInfo)
 		if err := m.repo.AddSnap(snapInfo); err != nil {
-			logger.Noticef("cannot add snap %q to interface repository: %s", snapInfo.Name(), err)
+			logger.Noticef("cannot add snap %q to interface repository: %s", snapInfo.InstanceName(), err)
```

Comment:
Suggestion for follow-up, make `snap.Info` a `Stringer` and use that, it could remove some of the explicitness from many places that don't need it.